controllers/chat/moveToAnon.js

```diff
@@ -54,102 +65,85 @@ const moveToAnon = async (req, res) => {
+        prevTargetUser?.anon_user_info_color_name;
+      newStateMemberWithAnonInfo[prevTargetUser.user_id].anon_user_info_emoji_code =
+        prevTargetUser?.anon_user_info_emoji_code;
+      newStateMemberWithAnonInfo[prevTargetUser.user_id].anon_user_info_emoji_name =
+        prevTargetUser?.anon_user_info_emoji_name;
+
+      await ChatAnonUserInfo.create({
+        channel_id: newChannel.id,
+        my_anon_user_id: prevTargetUser.user_id,
+        target_user_id: prevTargetUser.user_id,
+        anon_user_info_color_code: prevTargetUser?.anon_user_info_color_code,
+        anon_user_info_color_name: prevTargetUser?.anon_user_info_color_name,
+        anon_user_info_emoji_code: prevTargetUser?.anon_user_info_emoji_code,
+        anon_user_info_emoji_name: prevTargetUser?.anon_user_info_emoji_name
+      });
+    }
+
+    //set own user detail
+    let checkChatAnonUserInfo = await ChatAnonUserInfo.findOne({
+      where: {
+        channel_id: oldChannelId,
+        my_anon_user_id: req.userId,
+        target_user_id: prevTargetUser.user_id
+      }
+    });
+
+    if (checkChatAnonUserInfo) {
+      newStateMemberWithAnonInfo[req.userId].anon_user_info_color_code =
+        checkChatAnonUserInfo?.anon_user_info_color_code;
+      newStateMemberWithAnonInfo[req.userId].anon_user_info_color_name =
+        checkChatAnonUserInfo?.anon_user_info_color_name;
+      newStateMemberWithAnonInfo[req.userId].anon_user_info_emoji_code =
+        checkChatAnonUserInfo?.anon_user_info_emoji_code;
+      newStateMemberWithAnonInfo[req.userId].anon_user_info_emoji_name =
+        checkChatAnonUserInfo?.anon_user_info_emoji_name;
+
+      newStateMemberWithAnonInfo[req.userId].user.name =
+        'Anonymous ' + checkChatAnonUserInfo?.anon_user_info_emoji_name;
+      newStateMemberWithAnonInfo[req.userId].user.username =
+        'Anonymous ' + checkChatAnonUserInfo?.anon_user_info_emoji_name;
+    } else {
+      newStateMemberWithAnonInfo[req.userId].anon_user_info_color_code = anon_user_info_color_code;
+      newStateMemberWithAnonInfo[req.userId].anon_user_info_color_name = anon_user_info_color_name;
+      newStateMemberWithAnonInfo[req.userId].anon_user_info_emoji_code = anon_user_info_emoji_code;
+      newStateMemberWithAnonInfo[req.userId].anon_user_info_emoji_name = anon_user_info_emoji_name;
+
+      newStateMemberWithAnonInfo[req.userId].user.name = 'Anonymous ' + anon_user_info_emoji_name;
+      newStateMemberWithAnonInfo[req.userId].user.username =
+        'Anonymous ' + anon_user_info_emoji_name;
+
+      await ChatAnonUserInfo.create({
+        channel_id: newChannel.id,
+        my_anon_user_id: req.userId,
+        target_user_id: prevTargetUser.user_id,
+        anon_user_info_color_code: anon_user_info_color_code,
+        anon_user_info_color_name: anon_user_info_color_name,
+        anon_user_info_emoji_code: anon_user_info_emoji_code,
+        anon_user_info_emoji_name: anon_user_info_emoji_name
+      });
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This block of code is setting the own user details and creating an entry in `ChatAnonUserInfo` if necessary. Similar to the previous comment, it would be better to encapsulate this logic into a separate function for better readability and maintainability.

```diff
+ async function setOwnUserDetails(req, oldChannelId, prevTargetUser, newStateMemberWithAnonInfo, newChannel) {
+   let checkChatAnonUserInfo = await ChatAnonUserInfo.findOne({
+     where: {
+       channel_id: oldChannelId,
+       my_anon_user_id: req.userId,
+       target_user_id: prevTargetUser.user_id
+     }
+   });
+
+   if (checkChatAnonUserInfo) {
+     newStateMemberWithAnonInfo[req.userId].anon_user_info_color_code =
+       checkChatAnonUserInfo?.anon_user_info_color_code;
+     newStateMemberWithAnonInfo[req.userId].anon_user_info_color_name =
+       checkChatAnonUserInfo?.anon_user_info_color_name;
+     newStateMemberWithAnonInfo[req.userId].anon_user_info_emoji_code =
+       checkChatAnonUserInfo?.anon_user_info_emoji_code;
+     newStateMemberWithAnonInfo[req.userId].anon_user_info_emoji_name =
+       checkChatAnonUserInfo?.anon_user_info_emoji_name;
+
+     newStateMemberWithAnonInfo[req.userId].user.name =
+       'Anonymous ' + checkChatAnonUserInfo?.anon_user_info_emoji_name;
+     newStateMemberWithAnonInfo[req.userId].user.username =
+       'Anonymous ' + checkChatAnonUserInfo?.anon_user_info_emoji_name;
+   } else {
+     newStateMemberWithAnonInfo[req.userId].anon_user_info_color_code = anon_user_info_color_code;
+     newStateMemberWithAnonInfo[req.userId].anon_user_info_color_name = anon_user_info_color_name;
+     newStateMemberWithAnonInfo[req.userId].anon_user_info_emoji_code = anon_user_info_emoji_code;
+     newStateMemberWithAnonInfo[req.userId].anon_user_info_emoji_name = anon_user_info_emoji_name;
+
+     newStateMemberWithAnonInfo[req.userId].user.name = 'Anonymous ' + anon_user_info_emoji_name;
+     newStateMemberWithAnonInfo[req.userId].user.username =
+       'Anonymous ' + anon_user_info_emoji_name;
+
+     await ChatAnonUserInfo.create({
+       channel_id: newChannel.id,
+       my_anon_user_id: req.userId,
+       target_user_id: prevTargetUser.user_id,
+       anon_user_info_color_code: anon_user_info_color_code,
+       anon_user_info_color_name: anon_user_info_color_name,
+       anon_user_info_emoji_code: anon_user_info_emoji_code,
+       anon_user_info_emoji_name: anon_user_info_emoji_name
+     });
+   }
+ }

- //set own user detail
- let checkChatAnonUserInfo = await ChatAnonUserInfo.findOne({
-   where: {
-     channel_id: oldChannelId,
-     my_anon_user_id: req.userId,
-     target_user_id: prevTargetUser.user_id
-   }
- });
-
- if (checkChatAnonUserInfo) {
-   newStateMemberWithAnonInfo[req.userId].anon_user_info_color_code =
-     checkChatAnonUserInfo?.anon_user_info_color_code;
-   newStateMemberWithAnonInfo[req.userId].anon_user_info_color_name =
-     checkChatAnonUserInfo?.anon_user_info_color_name;
-   newStateMemberWithAnonInfo[req.userId].anon_user_info_emoji_code =
-     checkChatAnonUserInfo?.anon_user_info_emoji_code;
-   newStateMemberWithAnonInfo[req.userId].anon_user_info_emoji_name =
-     checkChatAnonUserInfo?.anon_user_info_emoji_name;
-
-   newStateMemberWithAnonInfo[req.userId].user.name =
-     'Anonymous ' + checkChatAnonUserInfo?.anon_user_info_emoji_name;
-   newStateMemberWithAnonInfo[req.userId].user.username =
-     'Anonymous ' + checkChatAnonUserInfo?.anon_user_info_emoji_name;
- } else {
-   newStateMemberWithAnonInfo[req.userId].anon_user_info_color_code = anon_user_info_color_code;
-   newStateMemberWithAnonInfo[req.userId].anon_user_info_color_name = anon_user_info_color_name;
-   newStateMemberWithAnonInfo[req.userId].anon_user_info_emoji_code = anon_user_info_emoji_code;
-   newStateMemberWithAnonInfo[req.userId].anon_user_info_emoji_name = anon_user_info_emoji_name;
-
-   newStateMemberWithAnonInfo[req.userId].user.name = 'Anonymous ' + anon_user_info_emoji_name;
-   newStateMemberWithAnonInfo[req.userId].user.username =
-     'Anonymous ' + anon_user_info_emoji_name;
-
-   await ChatAnonUserInfo.create({
-     channel_id: newChannel.id,
-     my_anon_user_id: req.userId,
-     target_user_id: prevTargetUser.user_id,
-     anon_user_info_color_code: anon_user_info_color_code,
-     anon_user_info_color_name: anon_user_info_color


<!-- This is an auto-generated comment by OSS CodeRabbit -->